lib/Backend/arm64/LowerMD.cpp

```diff
@@ -1298,24 +1181,45 @@ LowererMD::LowerEntryInstr(IR::EntryInstr * entryInstr)
         if (!IsSmallStack(stackAdjust) || useDynamicStackProbe)
         {
             unwindInfo->SetSavedScratchReg(true);
-            if (!usedRegs.Test(RegEncode[SP_ALLOC_SCRATCH_REG])) //If its a large stack and RegR19 is not already saved.
+            if (!usedRegs.Test(RegEncode[SP_ALLOC_SCRATCH_REG])) //If its a large stack and RegR15 is not already saved.
             {
                 // If it is not a small stack we have to call __chkstk.
-                // __chkstk has special calling convention and trashes R4
+                // __chkstk has special calling convention and is called with the amount to allocate divided by 16 in x15
                 // And if we're probing the stack dynamically, we need an extra reg to do the frame size calculation.
                 //
                 // Note that it's possible that we now no longer need a dynamic stack probe because
                 // m_localStackHeight may be encodable in Mod12. However, this is a chicken-and-egg
                 // problem, so we're going to stick with saving R4 even though it's possible it
```

Comment:
Comment needs updating. Mod12 is not the encoding limitation here and R4 is not the reg in question.